### DISCUSSION TITLE
cannot build ipa file using the yaml getting an error

### DISCUSSION THREAD

**Discussion by shmuelsoib** [codemagic-ci-cd/codemagic-docs#2019] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Hi, i set my my certificates and provisioning profile and tried to build a ipa file with the following yaml workflow:

  ios-workflow:
    name: iOS Workflow
    max_build_duration: 120
    environment:
      flutter: 3.7.11
      xcode: latest
      cocoapods: default
      ios_signing:
       distribution_type: app_store
       bundle_identifier: com.safeguard.safety
    scripts: 
      - |
        # set up local properties
      - flutter packages pub get
      - flutter pub get
      - xcode-project use-profiles
      - find . -name "Podfile" -execdir pod install \;
      - flutter build ipa --release --flavor safeguardprod --target lib/flavors/starter_files/main_safeguard_prod.dart --build-name=1.7.1 --build-number=121 -–export-options-plist=/Users/builder/export_options.plist
    artifacts:
      - build/ios/ipa/*.ipa
      - /tmp/xcodebuild_logs/*.log
      - flutter_drive.log
      
everything worked correctly except for the last command where i am getting the following error:
exportArchive: "Runner.app" requires a provisioning profile with the Near Field Communication Tag Reading and Push Notifications features.
i read in the documentation that this error could occur and followed the steps it said there:
1) to use xcode-project use-profiles which i have.
2) to use –export-options-plist=/Users/builder/export_options.plist which i have also
my provisioning profile includes the Near Field Communication Tag Reading and Push Notifications features
thank you for your time

### Expected results

an ipa file to be built

### Actual results

got the following error:
 exportArchive: "Runner.app" requires a provisioning profile with the Near Field Communication Tag Reading and Push Notifications features.

**Comment by dtrdic** [comment]
Hey @shmuelsoib, based on the error message, it seems you're still using a provisioning profile that doesn't have those capabilities enabled. Please try downloading it again from Apple Developer and then upload it to Codemagic.

**Reply by dtrdic** [reply]
but you can share your build Id as well, so I can take a closer look

**Reply by shmuelsoib** [reply]
this is the build id:
64f5dd20e0fc9fb6d02acf98

my  provisioning profiles have them enabled

**Reply by shmuelsoib** [reply]
this is the profile
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/54931488/59836114-a196-467c-ac15-d95235f45f25)

**Reply by dtrdic** [reply]
yep, i see.. ok before I continue investigating have you done this step that I advised above?
If you enabled those capabilities after you saw the error message and updated the profile, have you re-uploaded the profile with the new capabilities under code signing identities again?

**Reply by shmuelsoib** [reply]
these capabilities were set months ago for the profile. i downloaded it a few times and reuploaded it

**Reply by dtrdic** [reply]
have you enabled those capabilities inside Xcode under the project target and Signing & Capabilities ?
![Screenshot 2023-09-05 at 12 03 30](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/17b7d112-7e0e-45c8-855c-3af8c890df3a)

**Reply by shmuelsoib** [reply]
Yes I have

**Reply by shmuelsoib** [reply]
is that a problem?

**Reply by shmuelsoib** [reply]
@dtrdic have any ideas?

**Reply by dtrdic** [reply]
hey @shmuelsoib, open your `Runner.entitlements` file and check if you have these strings in array;
```
	<key>com.apple.developer.nfc.readersession.formats</key>
	<array>
		<string>NDEF</string>
		<string>TAG</string>
	</array>
```
also add this inside `Info.plist`
```
	<key>NFCReaderUsageDescription</key>
	<string>Explanation of why your app needs NFC access.</string>
```

**Reply by shmuelsoib** [reply]
@dtrdic  hi yes i have both of them

**Reply by shmuelsoib** [reply]
should i try something else? a different kind of build code?

**Reply by dtrdic** [reply]
hey @shmuelsoib can you please check if you have entitlements set on all flavors?
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/8f4f3002-3420-478c-914a-26a3ed020732)

I deleted a few comments from above, just to have cleaner conversation

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion due to a lack of activity. You can open the discussion again by replying to this message or opening a new discussion.
